packages/components/filters/filter-menu/src/filter-menu.tsx

```diff
@@ -0,0 +1,109 @@
+import { type ReactElement } from 'react';
+import DropdownMenu from '@commercetools-uikit/dropdown-menu';
+import { Footer } from './footer';
+import { Header } from './header';
+import { TriggerButton } from './trigger-button';
+
+export type TAppliedFilterValue = {
+  key?: string;
+  label: string | ReactElement;
+};
+
+export type TFilterMenuProps = {
+  /**
+   * unique identifier for the filter
+   */
+  filterKey: string;
+  /**
+   * formatted message to display the filter's name
+   */
+  label: string | ReactElement;
+  /**
+   * the input in which the user selects values for the filter
+   *
+   * NOTE / OPINIONS WELCOME IN PR COMMENTS:
+   * can either pass a react element and let the consuming app choose what input to use,
+   * or start defining different menu bodies based on props and only make predefined input
+   * types available, which is restrictive and much higher maintenance
+   */
+  filter: ReactElement;
```

Comment:
1. Having a `Field`-version of almost every input already takes a mental toll due to all the duplication, personally I don't need a `Filter`-version of every input on top. 
2. We're not enabling product-development if we'd have to whitelist every possible input first, we'd become a bottleneck. 

I'd let the consumer decide what input to use.

And what about using the `children` property for it?

```jsx
<FilterMenu we={take} care={of} repetitive={stuff} only>
   {`<-- complex, individual inputs -->`}
</FilterMenu>
```

Reply:
What's been asked for by product is a `Filters` "pattern". So the approach that I'm setting up for is that consuming apps pass an array of filter configuration objects to the parent `Filters`/`FiltersList` component, and that component then handles creating each `FilterMenu` based on the config.

So, passing it as `children` could work, but I went with `filter` because the component needs to be called something other than `children` when defined in the config.  If that makes sense?

Reply:
The rough idea is that wherever the overall `Filters`/`FiltersList` component is defined, the consuming app would pass down an array of objects that configure each `FilterMenu`, as pseudocoded below:

```jsx
const filterDefinitions = [
 {
  key: 'filterName',
  filter: <ComponentPassedFromConsumingApp props={...data/handlersFromConsumingApp}
 ...otherConfig
 }
]
<FiltersList
  filters={filterDefinitions}
  {...otherProps}
/>

```

`FiltersList` would then pass down `filterDefinitions.filter` to `FilterMenu`.  

So the approach is an "option 1" approach, in that the consuming app defines the component that is rendered in the `FilterMenu` body.  Does that help to explain what I meant in my previous comment?

As far as the naming of the prop, I'm fine with passing it as `children`.

I'm also open to suggestions for the key used in the config object passed to the parent `FiltersList` component.  We are expecting some sort of input, so `filterInput` works for me.  `menuBody` or `filterMenuBody` would also make sense.

Reply:
I see what you mean about not calling it children if you are setting it up in the filterDefinitions. If this is where you set it up, I like having a more descriptive name for it. I actually like incorporating `menuBody` somehow as that tells me where in the filter it's going to be used/rendered. Throwing out other options: `filterMenuInput` or `filterBodyInput` or `menuBodyInput`... 🤷‍♀️  sometimes I hate naming...